File: TP01/Alteracao_Aleatoria_Q4/AlteracaoAleatoria.java
import java.util.Random;
class AlteracaoAleatoria {

   public static boolean isFim(String s){
      return (s.length() >= 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
   }

   public static String Alteracao(String s, int contador) {
        int i, n = s.length();
        String resp = "";

        Random gerador = new Random();
        gerador.setSeed(4);
        
        char NovaLetra = ' ';
        char velhaLetra = ' ';
        for(int z = 0; z < contador; z++){            
            velhaLetra = (char) ('a' + Math.abs(gerador.nextInt() % 26));
            NovaLetra = (char) ('a' + Math.abs(gerador.nextInt() % 26));
        }
    
        for (i=0; i<n; i++) { 
            if(s.charAt(i) == velhaLetra){
               resp = resp + NovaLetra;
            }
            else{
             resp = resp + s.charAt(i);
            }
        }
 
      return resp;  
   }

   public static void main (String[] args){
      String[] entrada = new String[1000];
      int numEntrada = 0;

      //Leitura da entrada padrao
      do {
         entrada[numEntrada] = MyIO.readLine();
      } while (isFim(entrada[numEntrada++]) == false);
      numEntrada--;   //Desconsiderar ultima linha contendo a palavra FIM

      //Para cada linha de entrada, gerando uma de saida contendo o numero de letras maiusculas da entrada
      for(int i = 0; i < numEntrada; i++){
         MyIO.println(Alteracao(entrada[i],i+1));
      }
   }
}

File: TP01/Alteracao_Aleatoria_Recursivo_Q13/AlteracaoAleatoria.cpp
#include <stdlib.h>
#include <stdio.h>
#include <stdbool.h>
#include <string.h>


bool isFim(char* s)
{
   return (strlen(s) >= 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}

char * Alteracao(char* s, char VelhaLetra, char NovaLetra, int i)
{
    if(i < strlen(s))
    {
        if(s[i] == VelhaLetra)s[i] = NovaLetra;
        strcpy(s,Alteracao(s, VelhaLetra, NovaLetra, i + 1));
    }
    return s;
}


int main()
{
    char palavra[1000][1000];
    int i = 0;

    do                                             
    {
        fgets(palavra[i], 1000, stdin);
    } while (isFim(palavra[i++]) == false);       
    i--;

    srand(4);

    for(int c = 0; c < i; c++)
    {
        char VelhaLetra = (char)('a' + abs(rand() % 26));
        char NovaLetra = (char)('a' + abs(rand() % 26));
        printf("%s",Alteracao(palavra[c],VelhaLetra,NovaLetra,0));
    }
}
